src/runtime/pkg/containerd-shim-v2/service.go

```diff
@@ -794,7 +795,18 @@ func (s *service) Kill(ctx context.Context, r *taskAPI.KillRequest) (_ *ptypes.E
 		return empty, nil
 	}
 
-	return empty, s.sandbox.SignalProcess(spanCtx, c.id, processID, signum, r.All)
+	// kill(2) method can return ESRCH in certain cases, which is not handled by containerd cri server in container_stop.go.
+	// runc v2 translates ESRCH to ErrNotFound, so we'll handle similarly in Kata.
+	if err = s.sandbox.SignalProcess(spanCtx, c.id, processID, signum, r.All); err != nil &&
+		strings.Contains(err.Error(), "ESRCH: No such process") {
+		shimLog.WithFields(logrus.Fields{
+			"sandbox":   s.sandbox.ID(),
+			"container": c.id,
+			"exec-id":   r.ExecID,
+		}).Warn("signal encounters ESRCH, process already finished")
+		return empty, errors.Wrapf(errdefs.ErrNotFound, "process already finished")
```

Comment:
So, AFAICT CRIO doesn't know what to do with ErrNotFound, so while this'll make containerd happy, it'll break CRIO.

Perhaps the workaround at this point is to instead "swallow" the error here. Just print a warning (what you have there), and return empty, nil?